freeipa/src/test/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaTopologyServiceTest.java

```diff
@@ -168,6 +173,51 @@ void testUpdateReplicationTopology(int numNodes, int expectedSegmentsToAdd, int
         Mockito.verify(freeIpaClient, Mockito.times(expectedSegmentsToRemove)).deleteTopologySegment(Mockito.eq("domain"), Mockito.any());
     }
 
+    @MethodSource("testUpdateReplicationTopologyParameters")
+    @ParameterizedTest(name = "Run {index}: numNodes={0}, expectedSegmentsToAdd={1}, expectedSegmentsToRemove={2}")
+    void testUpdateReplicationTopologyWithRetry(int numNodes, int expectedSegmentsToAdd, int expectedSegmentsToRemove) throws FreeIpaClientException {
+        Stack stack = new Stack();
+        stack.setId(1L);
+        when(freeIpaClientFactory.getFreeIpaClientForStack(stack)).thenReturn(freeIpaClient);
+        Set<InstanceMetaData> imSet = new HashSet<>();
+        for (int i = 0; i < numNodes; i++) {
+            InstanceMetaData im = new InstanceMetaData();
+            im.setDiscoveryFQDN(String.format("ipaserver%d.example.com", i));
+            imSet.add(im);
+        }
+        when(instanceMetaDataService.findNotTerminatedForStack(anyLong())).thenReturn(imSet);
+        TopologySuffix caSuffix = new TopologySuffix();
+        caSuffix.setCn("ca");
+        TopologySuffix domainSuffix = new TopologySuffix();
+        domainSuffix.setCn("domain");
+        when(freeIpaClient.findAllTopologySuffixes()).thenReturn(List.of(caSuffix, domainSuffix));
+        List<TopologySegment> topologySegments1 = new LinkedList<>();
+        for (int i = 1; i < numNodes; i++) {
+            TopologySegment segment = new TopologySegment();
+            segment.setLeftNode("ipaserver0.example.com");
+            segment.setRightNode(String.format("ipaserver%d.example.com", i));
+            topologySegments1.add(segment);
+        }
+        List<TopologySegment> topologySegments2 = new LinkedList<>();
+        topologySegments2.addAll(topologySegments1);
+        when(freeIpaClient.findTopologySegments(Mockito.anyString()))
+                .thenReturn(topologySegments1)
+                .thenReturn(topologySegments1)
+                .thenReturn(topologySegments2)
+                .thenReturn(topologySegments2);
```

Comment:
Sorry, what is the point of `topologySegments2` in the first place when it is initialized identical to `topologySegments1`?

Reply:
no idea, I've copied the other test and adjusted the part only relevant to the new method

Reply:
Okay, whatever. I was just confused, but do not mind leaving this as is.